source/lib/operators/not.ts

```diff
@@ -11,10 +11,10 @@ export const not = <T extends Predicate>(predicate: T) => {
 		const fn = validator.validator;
 		const message = validator.message;
 
-		validator.message = (x: any) => `[NOT] ${message(x)}`;
-		validator.validator = (x: any) => !fn(x);
+		validator.message = (x: T) => `[NOT] ${message(x)}`;
```

Comment:
This is not of type `T`. The `T` type is a predicate, for instance `StringPredicate`. The value passed down in the message is the value that is tested, which could basically be anything. That's why, `any` :).